src/core/server/saved_objects/migrations/actions/es_errors.ts

```diff
@@ -21,3 +21,12 @@ export const isIncompatibleMappingException = ({ type }: estypes.ErrorCause): bo
 export const isIndexNotFoundException = ({ type }: estypes.ErrorCause): boolean => {
   return type === 'index_not_found_exception';
 };
+
+export const isClusterShardLimitExceeded = ({ type, reason }: estypes.ErrorCause): boolean => {
```

Comment:
NIT: I know it's covered by integration tests, but I would still add a UT to `src/core/server/saved_objects/migrations/actions/es_errors.test.ts`